lib/tapioca/generators/base.rb

```diff
@@ -0,0 +1,33 @@
+# typed: strict
+# frozen_string_literal: true
+
+# NOTE: Temporary measure to get pairity with existing behaviour. Thor should be abstracted out so Generators
+# do NOT need to know about the CLI interface.
+require "thor"
+
+module Tapioca
+  module Generators
+    class Base
+      extend T::Sig
+      extend T::Helpers
+
+      include Thor::Base # TODO: Remove me when logging logic has been abstracted.
+
+      sig { returns(Tapioca::Config) }
+      attr_reader :config
+
+      abstract!
+
+      sig { params(config: Tapioca::Config).void }
+      def initialize(config)
+        @config = config
+      end
+
+      sig { abstract.void }
```

Comment:
Let's not introduce this right now if it's only to override it with empty implementations.

Reply:
Or can't we pass all the options to the subclasses constructors so `generate` doesn't need different parameters and we can override it?

Reply:
Either of these sounds like what I was thinking of. But you're right, the abstract methods weren't needed yet, and was too early of an optimization. I removed them from this PR and can address these types of changes as I go back and clean up the code, etc.